src/components/overlay/overlay.ts

```diff
@@ -58,253 +47,85 @@ export class SbbOverlayElement extends SbbNegativeMixin(LitElement) {
    */
   @property({ reflect: true, type: Boolean }) public expanded = false;
 
-  /**
-   * Whether a back button is displayed next to the title.
-   */
+  /** Whether a back button is displayed next to the title. */
   @property({ attribute: 'back-button', type: Boolean }) public backButton = false;
 
-  /**
-   * This will be forwarded as aria-label to the close button element.
-   */
+  /** This will be forwarded as aria-label to the close button element. */
   @property({ attribute: 'accessibility-close-label' }) public accessibilityCloseLabel:
     | string
     | undefined;
 
-  /**
-   * This will be forwarded as aria-label to the back button element.
-   */
+  /** This will be forwarded as aria-label to the back button element. */
   @property({ attribute: 'accessibility-back-label' }) public accessibilityBackLabel:
     | string
     | undefined;
 
-  /**
-   * This will be forwarded as aria-label adn will describe the purpose of the dialog.
-   */
-  @property({ attribute: 'accessibility-label' }) public accessibilityLabel: string | undefined;
-
-  /*
-   * The state of the overlay.
-   */
-  private set _state(state: SbbOpenedClosedState) {
-    this.setAttribute('data-state', state);
-  }
-  private get _state(): SbbOpenedClosedState {
-    return this.getAttribute('data-state') as SbbOpenedClosedState;
-  }
-
-  private _ariaLiveRef!: SbbScreenReaderOnlyElement;
-  private _ariaLiveRefToggle = false;
-
-  /** Emits whenever the `sbb-overlay` starts the opening transition. */
-  private _willOpen: EventEmitter<void> = new EventEmitter(this, SbbOverlayElement.events.willOpen);
-
-  /** Emits whenever the `sbb-overlay` is opened. */
-  private _didOpen: EventEmitter<void> = new EventEmitter(this, SbbOverlayElement.events.didOpen);
-
-  /** Emits whenever the `sbb-overlay` begins the closing transition. */
-  private _willClose: EventEmitter = new EventEmitter(this, SbbOverlayElement.events.willClose);
-
-  /** Emits whenever the `sbb-overlay` is closed. */
-  private _didClose: EventEmitter<SbbOverlayCloseEventDetails> = new EventEmitter(
-    this,
-    SbbOverlayElement.events.didClose,
-  );
+  protected closeAttribute: string = 'sbb-overlay-close';
 
   /** Emits whenever the back button is clicked. */
   private _backClick: EventEmitter<any> = new EventEmitter(
     this,
     SbbOverlayElement.events.backClick,
   );
-
   private _overlayContentElement: HTMLElement | null = null;
-  private _overlayCloseElement?: HTMLElement;
-  private _overlayController!: AbortController;
-  private _openOverlayController!: AbortController;
-  private _focusHandler = new SbbFocusHandler();
-  private _scrollHandler = new SbbScrollHandler();
-  private _returnValue: any;
-
-  // Last element which had focus before the overlay was opened.
-  private _lastFocusedElement?: HTMLElement;
 
-  private _language = new SbbLanguageController(this);
-
-  /**
-   * Opens the overlay element.
-   */
+  /** Opens the component. */
   public open(): void {
```

Comment:
```suggestion
  public override open(): void {
```
I would mark the abstract implementation with the `override` keyword (might be a personal preference though)

Reply:
hmm.. since they are implementations from zero and not a rewriting of an existing method, i prefer not to add it..